control/state.py

```diff
@@ -222,6 +226,53 @@ def get_omap_version(self) -> int:
                 f" invalid number of values ({value_list}).")
             raise
 
+    def lock_omap(self, timeout = None):
+        got_lock = False
+
+        if timeout is None:
+            self.logger.warning(f"Can't lock OMAP file without a lock timeout")
+            return False
+
+        for i in range(1, self.omap_file_lock_retries):
+            try:
+                with rados.WriteOpCtx() as write_op:
+                    self.ioctx.lock_exclusive(self.omap_name, self.OMAP_FILE_LOCK_NAME, self.OMAP_FILE_LOCK_COOKIE, "OMAP file changes lock", timeout, 0)
+                    got_lock = True
+                    break
+            except rados.ObjectExists as ex:
+                self.logger.info(f"We already locked the OMAP file")
+                got_lock = True
+                break
+            except rados.ObjectBusy as ex:
+                self.logger.warning(f"Someone else locked the OMAP file, will try again in {self.omap_file_lock_retry_sleep_interval} seconds")
+                time.sleep(self.omap_file_lock_retry_sleep_interval)
+            except Exception as ex:
+                self.logger.error(f"Unable to lock OMAP file: {ex}. Exiting!")
+                raise
+
+        if not got_lock:
+            self.logger.error(f"Unable to lock OMAP file after {self.omap_file_lock_retries} retries. Exiting!")
+            raise Exception("Unable to lock OMAP file")
+
+        omap_version = self.get_omap_version()
+
+        if omap_version > self.version:
+            self.logger.warning(f"Local version {self.version} differs from OMAP file version {omap_version}, need to read the OMAP file")
+            self.unlock_omap()
+            raise OSError(11, "Unable to lock OMAP file, file not current", self.omap_name)
```

Comment:
Does the number 11 correspond to EAGAIN? How was this specific value chosen? Is it possible to use the Python symbol errno.EAGAIN instead?